### PR TITLE
[#600]Improve UX of wallet restore screen

### PR DESCRIPTION
**Pull request by gasstan**
This code review checklist is intended to serve as a starting point for the author and reviewer, although it may not be appropriate for all types of changes (e.g. fixing a spelling typo in documentation).  For more in-depth discussion of how we think about code review, please see [Code Review Guidelines](../blob/main/docs/CODE_REVIEW_GUIDELINES.md).

# Author

- [x] Self-review: Did you review your own code in GitHub's web interface? _Code often looks different when reviewing the diff in a browser, making it easier to spot potential bugs._
- [ ] Automated tests: Did you add appropriate automated tests for any code changes?
- [ ] Manual tests: Did you update the [manual tests](../blob/main/docs/testing/manual_testing) as appropriate? _While we aim for automated testing of the application, some aspects require manual testing. If you had to manually test something during development of this pull request, write those steps down._
- [ ] Code coverage: Did you check the code coverage report for the automated tests?  _While we are not looking for perfect coverage, the tool can point out potential cases that have been missed._ Code coverage can be generated with: `./gradlew check` for Kotlin modules and `./gradlew connectedCheck -PIS_ANDROID_INSTRUMENTATION_TEST_COVERAGE_ENABLED=true` for Android modules.
- [ ] Documentation: Did you update documentation as appropriate? (e.g [README.md](../blob/main/README.md), etc.)
- [x] Run the app: Did you run the app and try the changes?
- [x] Screenshots: Did you provide before and after UI screenshots in the description of this pull request?  _This is only applicable for changes that modify the UI._
- [x] Rebase and squash: Did you pull in the latest changes from the main branch and squash your commits before assigning a reviewer? _Having your code up to date and squashed will make it easier for others to review. Use best judgement when squashing commits, as some changes (such as refactoring) might be easier to review as a separate commit._

# Reviewer

- [x] Checklist review: Did you go through the code with the [Code Review Guidelines](../blob/main/docs/CODE_REVIEW_GUIDELINES.md) checklist?
- [x] Ad hoc review: Did you perform an ad hoc review?  _In addition to a first pass using the code review guidelines, do a second pass using your best judgement and experience which may identify additional questions or comments. Research shows that code review is most effective when done in multiple passes, where reviewers look for different things through each pass._
- [ ] Automated tests: Did you review the automated tests?
- [ ] Manual tests: Did you review the manual tests?
- [ ] How is code coverage affected? _We encourage you to compare coverage before and after changes and when possible, leaving it in a better place._
- [ ] Documentation: Did you review Docs, [README.md](../blob/main/README.md), and [Architecture.md](../blob/main/docs/Architecture.md) as appropriate?
- [x] Run the app: Did you run the app and try the changes? _While the CI server runs the app to look for build failures or crashes, humans running the app are more likely to notice unexpected log messages, UI inconsistencies, or bad output data. Perform this step last, after verifying the code changes are safe to run locally._

### CODE REVIEW

**Comment by ccjernigan** [comment]
Followup issue: https://github.com/zcash/secant-android-wallet/issues/675

File: ui-lib/src/main/java/co/electriccoin/zcash/ui/screen/restore/view/RestoreView.kt [hunk 1]
```diff
@@ -321,11 +340,23 @@ private fun Autocomplete(
 
         LazyRow(highlightModifier.testTag(RestoreTag.AUTOCOMPLETE_LAYOUT)) {
             items(it) {
-                Button(
-                    modifier = Modifier.testTag(RestoreTag.AUTOCOMPLETE_ITEM),
-                    onClick = { onSuggestionSelected(it) }
+                Surface( // TODO create chip
```
Comment (ccjernigan) [inline ui-lib/src/main/java/co/electriccoin/zcash/ui/screen/restore/view/RestoreView.kt:343]: What's this TODO comment for?  Is there a followup issue you want to file for this?

File: ui-lib/src/main/java/co/electriccoin/zcash/ui/screen/restore/view/RestoreView.kt [hunk 2]
```diff
@@ -321,11 +340,23 @@ private fun Autocomplete(
...
+                    color = MaterialTheme.colorScheme.secondary,
+                    shadowElevation = 8.dp
                 ) {
-                    Text(it)
+                    Text(
+                        text = it,
+                        style = MaterialTheme.typography.bodyLarge,
+                        color = MaterialTheme.colorScheme.onSecondary,
+                        modifier = Modifier
+                            .padding(start = 16.dp, end = 16.dp, top = 8.dp, bottom = 8.dp)
```
Comment (ccjernigan) [inline ui-lib/src/main/java/co/electriccoin/zcash/ui/screen/restore/view/RestoreView.kt:357]: We're starting to introduce a lot of padding values.

We should start to think about how we might account for these in a larger design system.  I don't think we have that architecture figured out just yet, but I want to note it because we'll probably need to revisit these in the near future.

@HonzaR do you have any further thoughts since you've been thinking about this?

**Review by ccjernigan — COMMENTED** [review]
This is a huge improvement!  Thanks for the effort that went into this 😄 

I filed one followup issue that we can address much later.  We should also get honza's input before merging this.

File: ui-lib/src/main/java/co/electriccoin/zcash/ui/screen/restore/view/RestoreView.kt [hunk 3]
```diff
@@ -159,88 +184,69 @@ private fun RestoreTopAppBar(onBack: () -> Unit, onClear: () -> Unit) {
     )
 }
 
-@Suppress("UNUSED_PARAMETER")
+// TODO [#672] https://github.com/zcash/secant-android-wallet/issues/672
```
Comment (HonzaR) [inline ui-lib/src/main/java/co/electriccoin/zcash/ui/screen/restore/view/RestoreView.kt:187]: A really minor thing - we agreed to use a bit more detailed two-lined TODO comment with also a description of the linked issue, so we don't have to open the link. E.g. like this [one](https://github.com/zcash/secant-android-wallet/blob/381af575efbe87b9d62c7a410fadab4ded1b1a1a/gradle.properties#L146).

**Review by HonzaR — COMMENTED** [review]
It looks and works really great! I left one minor comment on the code. I also user tested it and I found few corner cases. Here are notes/questions on it:

- Why do we now use two titles here? I mean the one _Create a wallet_ in toolbar and the second bold one _Wallet import_. 
- I'd like to distinguish visually already confirmed chips from those in the suggestion box (e.g. with a slightly different background color). What do you think? 
Screenshot_20221123_153121

- I was able to reproduce a UI state in which a user can't see what he/she typing. We can e.g. make the whole bottom part (above keyboard) scrollable so it's not deformed then? 
Screenshot_20221123_152957

File: ui-lib/src/main/java/co/electriccoin/zcash/ui/screen/restore/view/RestoreView.kt (hunk 2 above)
Comment (gasstan) [reply ui-lib/src/main/java/co/electriccoin/zcash/ui/screen/restore/view/RestoreView.kt:357]: I 100% agree, we could take it a step. further and create our own padding Modifiers . what do you think?

File: ui-lib/src/main/java/co/electriccoin/zcash/ui/screen/restore/view/RestoreView.kt (hunk 1 above)
Comment (gasstan) [reply ui-lib/src/main/java/co/electriccoin/zcash/ui/screen/restore/view/RestoreView.kt:343]: no sorry, I forgotten about this. I will fix it

File: ui-lib/src/main/java/co/electriccoin/zcash/ui/screen/restore/view/RestoreView.kt (hunk 2 above)
Comment (HonzaR) [reply ui-lib/src/main/java/co/electriccoin/zcash/ui/screen/restore/view/RestoreView.kt:357]: Yes, could be. Good idea. My plan is to prepare a separate Dimens class, which will hold needed spacing values, and also enables us to switch between these predefined group of spacing for small/normal/large screens as well as for portrait/landscape, etc.

File: ui-lib/src/main/java/co/electriccoin/zcash/ui/screen/restore/view/RestoreView.kt (hunk 3 above)
Comment (gasstan) [reply ui-lib/src/main/java/co/electriccoin/zcash/ui/screen/restore/view/RestoreView.kt:187]: > * Why do we now use two titles here? I mean the one _Create a wallet_ in toolbar and the second bold one _Wallet import_.

 i added the second title to reflect wireframes, but i agree that one is probably enough...

> * I'd like to distinguish visually already confirmed chips from those in the suggestion box (e.g. with a slightly different background color). What do you think?

 i think this is a good idea!

> * I was able to reproduce a UI state in which a user can't see what he/she typing. We can e.g. make the whole bottom part (above keyboard) scrollable so it's not deformed then?

i think that if we lock screen orientation to Portrait it will resolve the issue, I'm not sure if we need to support landscape orientation

@geffenz  any thoughts on this?

Comment (ccjernigan) [reply ui-lib/src/main/java/co/electriccoin/zcash/ui/screen/restore/view/RestoreView.kt:187]: Locking to portrait hides significant Android lifecycle bugs during development and therefore we should continue suffering this pain of screen rotation for now.

Comment (HonzaR) [reply ui-lib/src/main/java/co/electriccoin/zcash/ui/screen/restore/view/RestoreView.kt:187]: Yes, agree. We should be prepared for landscape mode from the beginning. Once the MVP is out and things stabilized, we can move towards having a separate Composes for the landscape orientation. Until then, we're good with these mixed orientation screens.

Comment (geffenz) [reply ui-lib/src/main/java/co/electriccoin/zcash/ui/screen/restore/view/RestoreView.kt:187]: I absolutely agree on both suffering the pains now and later creating different Composes per orientation!

**Comment by gasstan** [comment]
here are changes Geffen requested:

1. Remove redundant header
2. fix alignment of chip indexes
3. hide suggestions when showing error
4. we also wanted to remove the number row from the keyboard but changing the keyboard type just replaces the number row with the suggestions row. Any thoughts on this?

**Comment by gasstan** [comment]
I think we resolved all issues for this PR or are there any changes that you guys want me to do? @HonzaR @ccjernigan

**Comment by ccjernigan** [comment]
I just need to run it again locally since there have been changes since I last ran it.

**Review by ccjernigan — APPROVED** [review]
